tests/nn/data_parallel/test_fsdp_memory.py

```diff
@@ -147,13 +141,14 @@ def _distributed_worker(gpu_id, world_size, with_fsdp, with_checkpoint, filename
         get_cur_mem(gpu_id, results, f"iter {iteration}: done")
 
     assert results == expected, f"{results} but expected {expected}"
+    dump_all_tensors(gpu_id)
 
     teardown()
 
 
 @skip_if_single_gpu
 @pytest.mark.parametrize("ckpt", ["no_ckpt", "ckpt"])
-@pytest.mark.parametrize("fsdp", ["ddp", "fsdp"])
+@pytest.mark.parametrize("fsdp", ["ddp", "fsdp", "fsdp_amp_default", "fsdp_amp_compute_dtype32"])
```

Comment:
curious about why we have ddp as a parameter? sanity check?

Reply:
yes, when we don't use p.grad=None, the memory for ddp and fsdp are very different. It is useful for checking and detect future changes in ddp. it is similar to when we compare numerical outputs of ddp and fsdp but here it is not directly.